provider/spark.go

```diff
@@ -136,17 +136,21 @@ func (s3 SparkS3FileStore) Type() string {
 }
 
 func (s3 SparkS3FileStore) PathWithPrefix(path string, remote bool) string {
-	noS3Prefix := !strings.HasPrefix(path, "s3a://")
+	pathContainsS3Prefix := strings.HasPrefix(path, s3aPrefix)
```

Comment:
I think we need to make a struct called Path that handles all this stuff across all the different files. Thoughts @sdreyer ? We can do it later but this just seems like a mess, I can see why path keeps being an issue.

Reply:
agreed